api/users.go

```diff
@@ -293,3 +293,53 @@ func (api *API) PasswordResetHandler(ctx context.Context, w http.ResponseWriter,
 
 	return models.NewSuccessResponse(nil, http.StatusAccepted, nil), nil
 }
+
+func (api *API) getGroupsForUser(listOfGroups []*cognitoidentityprovider.GroupType, userId models.UserParams) ([]*cognitoidentityprovider.GroupType, error) {
+	firstTimeCheck := false
+	var nextToken string
+	for {
+		if firstTimeCheck && nextToken == "" {
+			break
+		}
+		firstTimeCheck = true
+
+		userGroupsRequest := userId.BuildListUserGroupsRequest(api.UserPoolId, nextToken)
+		userGroupsResponse, err := api.CognitoClient.AdminListGroupsForUser(userGroupsRequest)
+		if err != nil {
+			return nil, err
+		}
+
+		listOfGroups = append(listOfGroups, userGroupsResponse.Groups...)
+		nextToken = ""
+		if userGroupsResponse.NextToken != nil {
+			nextToken = *userGroupsResponse.NextToken
+		}
+	}
+	return listOfGroups, nil
+}
+
+//ListUserGroupsHandler lists the users in the user pool
+func (api *API) ListUserGroupsHandler(ctx context.Context, w http.ResponseWriter, req *http.Request) (*models.SuccessResponse, *models.ErrorResponse) {
+
+	vars := mux.Vars(req)
+	userID := models.UserParams{ID: vars["id"]}
+	listofgroupsInput := []*cognitoidentityprovider.GroupType{}
+	finalUserResponse := cognitoidentityprovider.AdminListGroupsForUserOutput{}
+	listusergroups := models.ListUserGroups{}
+
+	listofGroupsOutput, err := api.getGroupsForUser(listofgroupsInput, userID)
+	if err != nil {
+		cognitoErr := models.NewCognitoError(ctx, err, "Cognito ListofUserGroups request from list user groups endpoint")
+		if cognitoErr.Code == models.NotFoundError {
+			return nil, models.NewErrorResponse(http.StatusBadRequest, nil, cognitoErr)
```

Comment:
Just checking if this is the right status code? Should it be bad request or not found?